### ISSUE TITLE
reconcile loop fails after configuring project matrix authorization strategy if the entries are high in number

### ISSUE THREAD

**Issue by salluvada** [jenkinsci/kubernetes-operator#290]
I am trying configure the authorization for jenkins using matrix-auth plugin and I noticed that when there are several entries in the configuration yaml (200 entries) for ProjectMatrixAuthorization, the configuration is not successful because of the error below. I do notice that when I have a smaller number of entries (10 entries), the configuration completes successfully. The jenkins operator user name I am using in the operator credentials secret has been given administrator rights in the configuration for ProjectMatrixAuthorization.

jenkins operator logs:
```
2020-03-17T10:11:53.274Z        INFO    controller-jenkins      groovy/groovy.go:145    user-casc ConfigMap 'project-matrix-auth-configuration-casc' name '1-project-matrix-auth.yaml' running groovy script    {"cr": "scdmkubejenkins"}
2020-03-17T10:11:53.318Z        WARN    controller-jenkins      jenkins/jenkins_controller.go:171       Reconcile loop failed: invalid status code '414', logs '<h1>Bad Message 414</h1><pre>reason: URI Too Long</pre>'        {"cr": "scdmkubejenkins"}
2020-03-17T10:12:04.034Z        INFO    controller-jenkins      groovy/groovy.go:145    user-casc ConfigMap 'project-matrix-auth-configuration-casc' name '1-project-matrix-auth.yaml' running groovy script    {"cr": "scdmkubejenkins"}
2020-03-17T10:12:04.074Z        WARN    controller-jenkins      jenkins/jenkins_controller.go:171       Reconcile loop failed: invalid status code '414', logs '<h1>Bad Message 414</h1><pre>reason: URI Too Long</pre>'        {"cr": "scdmkubejenkins"}
2020-03-17T10:12:25.040Z        INFO    controller-jenkins      groovy/groovy.go:145    user-casc ConfigMap 'project-matrix-auth-configuration-casc' name '1-project-matrix-auth.yaml' running groovy script    {"cr": "scdmkubejenkins"}
2020-03-17T10:12:25.080Z        WARN    controller-jenkins      jenkins/jenkins_controller.go:171       Reconcile loop failed: invalid status code '414', logs '<h1>Bad Message 414</h1><pre>reason: URI Too Long</pre>'        {"cr": "scdmkubejenkins"}
2020-03-17T10:13:06.662Z        INFO    controller-jenkins      groovy/groovy.go:145    user-casc ConfigMap 'project-matrix-auth-configuration-casc' name '1-project-matrix-auth.yaml' running groovy script    {"cr": "scdmkubejenkins"}
2020-03-17T10:13:06.721Z        WARN    controller-jenkins      jenkins/jenkins_controller.go:171       Reconcile loop failed: invalid status code '414', logs '<h1>Bad Message 414</h1><pre>reason: URI Too Long</pre>'        {"cr": "scdmkubejenkins"}
```

jenkins master logs:
```
2020-03-17 10:11:43.228+0000 [id=13]    WARNING o.eclipse.jetty.http.HttpParser#parseLine: URI is too large >8192
2020-03-17 10:11:43.858+0000 [id=16]    WARNING o.eclipse.jetty.http.HttpParser#parseLine: URI is too large >8192
2020-03-17 10:11:44.553+0000 [id=16]    WARNING o.eclipse.jetty.http.HttpParser#parseLine: URI is too large >8192
2020-03-17 10:11:47.593+0000 [id=16]    WARNING o.eclipse.jetty.http.HttpParser#parseLine: URI is too large >8192
```

**Comment by Nuru** [comment]
I have a similar issue with Configuration-as-Code. Is it also fixed by #290? 
```
    controller-jenkins      jenkins/jenkins_controller.go:171       Reconcile loop failed: invalid status code '414', logs '<h1>Bad Message 414</h1><pre>reason: URI Too Long</pre>'        {"cr": "jenkins"}
```

**Comment by tomaszsek** [comment]
Fixed in https://github.com/jenkinsci/kubernetes-operator/releases/tag/v0.4.0
